tests/unit/s2n_client_signature_algorithms_extension_test.c

```diff
@@ -71,33 +71,6 @@ int main(int argc, char **argv)
         s2n_connection_free(server_conn);
     };
 
-    /* Test that unknown TLS_EXTENSION_SIGNATURE_ALGORITHMS values are ignored and negotiation fails */
-    {
-        struct s2n_sig_scheme_list sig_hash_algs = {
-            .iana_list = { 0xFF01, 0xFFFF },
-            .len = 2,
-        };
-        struct s2n_connection *conn;
-        EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_SERVER));
-
-        struct s2n_stuffer signature_algorithms_extension = { 0 };
-        EXPECT_SUCCESS(s2n_stuffer_alloc(&signature_algorithms_extension, 2 + (sig_hash_algs.len * 2)));
-        POSIX_GUARD(s2n_stuffer_write_uint16(&signature_algorithms_extension, sig_hash_algs.len * 2));
-        for (size_t i = 0; i < sig_hash_algs.len; i++) {
-            POSIX_GUARD(s2n_stuffer_write_uint16(&signature_algorithms_extension, sig_hash_algs.iana_list[i]));
-        }
-
-        /* If only unknown algorithms are offered, expect choosing a scheme to fail for TLS1.3 */
-        conn->actual_protocol_version = S2N_TLS13;
-        EXPECT_SUCCESS(s2n_client_signature_algorithms_extension.recv(conn, &signature_algorithms_extension));
-        EXPECT_EQUAL(conn->handshake_params.client_sig_hash_algs.len, sig_hash_algs.len);
-        EXPECT_FAILURE(s2n_choose_sig_scheme_from_peer_preference_list(conn, &conn->handshake_params.client_sig_hash_algs,
-                &conn->handshake_params.server_cert_sig_scheme));
```

Comment:
This isn't actually what is happening in this test. If only unknown algorithms are offered, TLS1.3 should choose a fallback option instead of failing-- see the tests in s2n_signature_algorithms_test.

What's happening here is actually that the default config is being used, so there are no certificates configured. With no certificates, there are no valid signature schemes to choose from. The peer's preferences aren't even considered.

I just removed this test rather than fixing it. We already test this correctly in s2n_signature_algorithms_test, and it isn't really relevant to the extension's parsing.